### PR TITLE
SSPROD-33102: Pass provider alias in resource

### PR DESCRIPTION
**Pull request by raghavanandan**
This PR is to pass `provider_alias` in the resource so it can be used in the provider for cloud account creation.

### CODE REVIEW

File: modules/services/service-principal/outputs.tf [hunk 1]
```diff
@@ -17,6 +17,11 @@ output "service_principal_app_display_name" {
     description = "Display name of the Application created"
 }
 
+output "service_principal_app_id" {
```
Comment (mdkulkarni15) [inline modules/services/service-principal/outputs.tf:20]: do we need this? I don't think we have application_id exported that is replaced by client_id now

**Review by mdkulkarni15 — APPROVED** [review]
Other than the comment looks good.

File: modules/services/service-principal/outputs.tf (hunk 1 above)
Comment (raghavanandan) [reply modules/services/service-principal/outputs.tf:20]: I wonder how we were accessing this value in `app_id = module.project-posture.service_principal_app_id`. Terraform threw an error without having this variable defined.

Comment (mdkulkarni15) [reply modules/services/service-principal/outputs.tf:20]: that needs to change too.

**Review by mdkulkarni15 — APPROVED** [review]

**Pull request merged**